PIG 24: Authorship Policy (#3970)
gammapy/gammapy: docs/development/pigs/pig-024.rst

Given the fact that Gammapy is licensed under a 3-clause BSD style license (see gammapy/LICENSE.rst), Gammapy can be used
and even modified for a science project. For this modified version, the proposed authorship policy of this PIG is not
applicable but the general citation scheme should be applied.

This PIG is structured as follows: a reminder of our general citation scheme that was up-to-now only given orally and on
our web pages; then, the authorship policy is given for each of the products associated with Gammapy, namely the intermediate
releases, the Long Term Support (LTS) releases, the general Gammapy papers, and the contributions to the conference.

2. Reminder of the citation scheme
==================================

When Gammapy is used for any publication, contribution to conferences or software, authors should cite properly Gammapy.
It is asked to cite the DOI of the latest Gammapy paper.

.. note::
    As the Gammapy community fully supports Open Science, we strongly encourage authors using Gammapy to follow
    the FAIR4RS principles and to allow the reproducibility of their results. As consequence, we suggest always mentioning

>     the `Zenodo <https://zenodo.org/>`_ DOI or the `HAL <https://hal.archives-ouvertes.fr/>`_ identifier (associated with

Reviewer: This is confusing. I would give Zenodo by default and mention other options such as HAL.

Author: Zenodo does not provide a long-term archive. One should not forget that! Pushing for Zenodo is then not necessary our task. This is a political message.
Giving the choice to people is more neutral

    the `Software Heritage <https://www.softwareheritage.org/>`_ archive) of the used release of Gammapy.

This scheme and these two references are precise on our web pages.

3. Authorship policy
====================

The Gammapy references contain a list of authors, or `contributors`, that requires to be updated with time according to a
general policy. This section defines what is a contributor to Gammapy and the policy for each type of product.

Definition of a `Contributor`
-----------------------------

Contributing to Gammapy can be made in different ways:

- contributing to gammapy source code,
- contributing to our documentation (rst pages, docstring, gammapy-webpage),

> - contributing to our DevOps environment (CI, GitHub configurations, gammapy-docs, release processes, etc),

Reviewer: Maybe CI, docs and package deployment

> - contributing to our associated projects (gammapy-benchmarks, gammapy-data, gammapy-extra, gammapy-cat, gammapy-recipes),

Reviewer 2: I think here we should b encore specific:
- Either list all repositories by name
- Or, if applicable, a general criterion such as "all repositories under the Gammapy GitHub organisation"

Reviewer: Does contributing to gammapy-recipe make one a gammapy contributor? Does it mean contributing to the maintenance or contributing content?

Author: Maybe gammapy-recipe is not yet a contribution... So, with all other comments, the list is 'gammapy-benchmarks, gammapy-data'.
Should we keep gamma-cat or not???

> - contributing to the maintenance of our repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),

Reviewer 3: Maybe this needs some rephrasing.
I don't think it was meant as maintaining the platforms but rather maintain the gammapy uploads on these platforms. For example someone responsible for maintaining gammapy record on Zenodo (updating metadata if needed, requesting to be added to communities...)

> - contributing to the maintenance of our repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),

Reviewer: I suppose this goes rather into the devops category, e.g. as part of the release process.

> - contributing to the maintenance of our repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),

Author: Indeed, one can merge these tasks (ie maintain the gammapy uploads on these platforms., etc) into the DevOps!

> - leading the precise tasks identified by the Lead Developers,

Reviewer 2: This sentence is not clear to me. I guess it goes into the direction of "maintainer role" in the Gammapy project?

Reviewer: This is a bit unclear. If this PIG defines two types of roles: contributors and maintainers, then this should be part of the second, right?

Author: The sentence is written such that the L.D. have some freedom.
The PIG of tasks, module responsibles, etc.. is not written. This is why I put this sentence per anticipation. 
If you have an other formulation, you're welcome;)

Reviewer 4: This sentence is also too vague to be useful for me. If you want to go into the direction of long-term planning / steering the project, maybe this is clearer:

> contributing to the long-term project roadmap and over-all goals

?

> - organizing and leading hands-on sessions or schools,

Author: One has to define what is a contributor in order to make an author list!
A PR is easy to track, whatever the repo. Some DeVOps actions are not associated to a PR, this is more complicated to track.
But, if we have an official list of 'tasks', the associated people will be easily added to the contributor list.

> - organizing and leading hands-on sessions or schools,

Reviewer 2: I think there is a value in simplicity. Maybe it is simpler to mostly boil down to:
- Official role in the Gammapy project (listed here: https://gammapy.org/team.html)
- Present as author in the main Gammapy git history (without threshold)
- If some contribution is not covered by this, the person could be just required to "opt in"  and submit a PR adding their name to `CITATION.cff`. In which case point two (being part of the git history) holds again.
 
The team / CC / lead devs will then decide to approve the PR justified by the long list of possible contributions you mentioned. But I would add the "opt in" barrier. In 99% of the case it will be uncontroversial to add a person or not...

> - organizing and leading hands-on sessions or schools,

Reviewer 2: Keeping track of all contributions (beside from official roles and git history) from our side is to much of a burden. So one further argument for the "opt in" system, with team approval (the exact process needs to be decided...)

> - organizing and leading hands-on sessions or schools,

Author: The opt-in option, in addition to the PR and official roles, is very good... and lead to a great simplification of the policy. I like this!

> - organizing and leading hands-on sessions or schools,

Reviewer 4: I'd also support an opt-in approach for people who want to be on the author list but have only made these "indirect" contributions.

> - organizing and leading hands-on sessions or schools,

Author: This section is about the definition of a contributor.
In the next section, we define what are the authors of the products. And it is mentioned that the list of authors can be extended.

> - organizing and leading hands-on sessions or schools,

Reviewer 4: Yes, my point is: instead of defining a complex scheme of what a "contributor" is, it might be easier to have a very simple schema with a few things that make you definitely a contributor (actual commits, council, reviews) and then say

> For other forms of contributions, a user can ask the lead developers or the council to be included in the list of contributors

> - managing our social media accounts,

Reviewer: Maybe this along the webpage belongs to the maintainers tasks and should not be listed per-se here.

Author: Yes, but I do not have yet the maintainer tasks and the associated list of people;)

> - and, de-facto, the members of the Coordination Committee.

Author: So, one should make a difference between a technical contributor and an organisational/funding contributor? In Astropy, the member of the CC are in the list of authors...
So, one can say in this PIG: 'authors = (Technical) Contributors + CC'?

> - and, de-facto, the members of the Coordination Committee.

Reviewer 2: Keep in mind that people also get credit for their official roles in the project from the website: https://gammapy.org/team.html

> 

Reviewer 5: Since the contribution policy is quite liberal, maybe people doing PR reviews can also be counted?

Reviewer 3: That is indeed a very useful contribution to any open-source code.

Reviewer 2: I agree, however technically I have no good proposal of how to handle this in practice. How would we maintain a list of reviewers, that are not code contributors and do not have a git record?

Author: This is my main issue, to identify people doing the review. However, concretely, the people making the reviews are at >99.9%:
- the Lead Dev, that make PR so they are in
- the core dev team, that make PR so they are in
Do we have reviewers that never made a PR???

Reviewer 2: > Do we have reviewers that never made a PR???

Yes, e.g. here https://github.com/gammapy/gammapy/pull/3752#discussion_r835538313

Reviewer 3: > Again maybe replace by "opt in" system. Any person could propose to become a Gammapy author, if they think they should be. Decisions are taken by the team on a one by one basis. If a person has done many reviews the decision will be simple, transparent and obvious. However from the other perspective: finding all people and add them to the other list is a lot of work and we will miss those...

I completely agree with you here. People that think should be included in the author list should open a PR to add themselves to the author list, centralized now in the `CITATION.cff`. But at least from the policy, they know they are welcome to do so...

Reviewer 6: We are missing the gammapy CC at this point, no?

Author: One trend of opinions was to have the (GitHub) contributors for the code releases and the contributors + people with tasks in the project (e.g. the CC) for the LTS papers.
Should we extend that to the code releases?

Reviewer 4: At least code reviews are very directly connected to code releases

> Any contribution on these topics renders the user as an official Gammapy contributor.

Reviewer: Any contribution, really? 
I would not keep the sentence.

Author: Either you fix a threshold or not. Indeed, this sentence means 'no threshold'. This is why it is useful, you get it.
If you propose thresholds, can you make a proposal?

Reviewer 2: Yes, defining a threshold is tricky to impossible. I think the only practical threshold is >0.

.. important::
    We strongly recommend that the contributors use always the same GitHub account, to have a unique
    `ORCID <https://orcid.org/>`_ identifier and store it into your GitHub user profile, and inform the developers

>     of any change of affiliation.

Reviewer 2: Changes in GIthub user names are maintained in https://github.com/gammapy/gammapy/blob/master/.mailmap It would be good specify whether the responsibility of maintenance is on the Gammapy project side or contributor side. The latter would mea:  "if you don't make sure the info is correct, your contributions will not attributed correctly". Maybe it's both...

Reviewer 3: I think the list of contributors should be maintained by gammapy, in a metadata file (many choices: mailmap, citation.cff, codemeta.json, zenodo.json, tributors...) with the responsability falling on contributors to make sure their affiliation is correct, and update it through PR if not.
The CI could check if the author of a PR is already listed in this metadata (prob some dev needed ther)

Author: We are mixing 2 thinks, I think...
The list of contributors should indeed manager by Gammapy.
However, the user profile should be managed by them-self. They contain private information and as an open project (and not a closed collaboration), I feel very uncomfortable to deal with users data. This is why I put this important recommendation!

As Gammapy is an open project, we encourage users to provide feedback on the usage of Gammapy, on features, on issues

> via our communication channels.

Author: done

Definition of the `Maintainer`
------------------------------

In the Open Source community, a maintainer is generally mentioned in the metadata associated with a release and has to be mentioned.
Usually, a software maintainer or package maintainer is one or more people who build source code into a binary package
for distribution, commit patches, or organize code in a source repository.

For Gammapy, by default, the maintainers are the Lead Developers. If in the future a task is dedicated to the creation

> of a release, the maintainer will be the person in charge of this task.

Reviewer 2: Note that this is associated to the PyPi and GitHub access rights as well.

Reviewer 4: > GitHub does not provide metadata.

That is not entirely true, e.g. for some definition of maintainer, there is code-owners:
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Releases
--------

Each Gammapy release should be associated with an updated list of authors. The list of authors is composed by
people who contributed as listed in the previous sub-section to the current, or in the previous release in
order to account for long contributions (e.g. management of depreciated features, improvement/debugging of features,
etc). Any additional author can be inserted into the list for its special contribution when requested by the Gammapy

> Coordination Committee.

Reviewer: and/or lead dev.

Author: done

The order of the authors is first 'The Gammapy Collaboration', the Lead Developers, a past Lead Developer in case of
change between two releases, followed by the list of contributors in alphabetical order.

This policy is applied to any release after the LTS V1.0. For the releases made before, a hand-made list is used.
After the mention of 'The Gammapy Collaboration', the current Lead Developers will be in first, then the past Lead
Developers, followed by the list of contributors in alphabetical order.

Long Term Support releases
--------------------------

> 

Author: In this PIG, I have not written that Gammapy will always make a paper for an LTS. Maybe it is out-of-scope of this PIG ?

> 

Reviewer 4: You wrote that people should cite the paper belonging to the latest LTS release, which implies that there always is such a paper

> 

Reviewer 2: We are preparing the paper for v1.0, so this will be there for sure. But indeed if there is no paper v2.0 then people should continue citing the one for v1.0 instead. 

In general a small paper every two years should be possible to write and is fair to developers to collect academic credit as well. But there are thing out of our control so, e.g. maybe the paper is not accepted, so I agree we should rephrase to cite the latest Gammapy paper instead (whichever it is...).

> 

Reviewer 2: >yes, this is what we have foreseen and decided.

I fully agree this is what we should aim for, but I'm not sure we can commit to it? E.g. a paper could be not accepted for publication (it's unlikely but anyway...)

> 

Author: the new sentence is :+1: 
`When Gammapy is used for any publication, contribution to conferences or software, authors should properly cite Gammapy. It is asked to cite the DOI of the used Gammapy version as well as the associated paper (e.g. the latest LTS release).`

AND, few lines after:
`This citation scheme with these **two references** will be given on our web pages.
`

> The list of authors is composed of the union of all the contributors of the releases realised up to the last LST

Author: Yes, indeed! Any other proposal?

> The list of authors is composed of the union of all the contributors of the releases realised up to the last LST

Reviewer 2: No, I just wanted to have that clarified. I think it is fair to update the author list every 2 years. It is a similar process for many larger collaborations...

"Lifetime" contributors (water ever that means...) who left could still be acknowledged in the acknowledgment section. Or we introduce something like https://scikit-image.org/#emeritus-developers on our webpage.

change since the last LST, followed by the list of contributors in alphabetical order.

For the first LST release, the V1.0, all contributors from the beginning of the project will be co-authors. After the
mention of 'The Gammapy Collaboration', the Lead Developers will be in first, then the past Lead Developers since the
beginning of the project, followed by the list of contributors in alphabetical order.

General Gammapy publications
----------------------------

This product aims to describe the project and/or the library as a whole and is submitted to peer-review journals. As it
targets a wide community, the following scheme is used.

A/ List of authors

A.1/ By default, 'The Gammapy Collaboration' is mentioned first, then the Lead Developers, then the past Lead Developers since the
last general Gammapy publication, then the list of contributors of each release since the last Gammapy publication.

A.2/ If the editorial rules of the targeted journal permit it, the scheme used by the Astropy project
(e.g. `Astropy V2.0 <https://arxiv.org/pdf/1801.02634.pdf>`_) should be used in priority:

> - 'The Gammapy Collaboration', and the list of primary paper contributors in order of contribution agreed per consensus with the development team and the Coordination Committee, as '(Primary Paper Contributors)'

Reviewer: I'd rather go for gammapy team rather than collaboration.

Reviewer 2: +1 For "Gammapy Team" or "Gammapy Developers"

Author: ok

the paper in order of contribution, the members of the Gammapy Project Coordination Committee,
and contributors to the Gammapy Project in alphabetical order. The position in the author list does not
correspond to contributions to the Gammapy Project as a whole. A more complete list of contributors
to the core package can be found in the package repository and at the Gammapy team webpage."

C/ The corresponding author is the 'Gammapy Coordination Committee', associated with its usual mailing list.

D/ Acknowledgement

One has the freedom to precisely mention acknowledgements associated with the publication. 
In practice, it is recommended to precise the grants or fellowships given to some authors.
One should in any case always acknowledge the Astropy project, to which we are affiliated.

Contributions in conferences
-----------------------------

The section concerns any contribution in conferences (a talk, a poster and their associated proceedings) related to
the Gammapy project itself. It does not concern a technical or scientific work that uses Gammapy as an open library.
In this case, the standard citation scheme should be used.

> 

Author: I put that in the implementation!

> As the length of the author list is generally a constrain, the author list can be reduced to the shortlist of authors,

Reviewer 5: Any proposals on choosing this short author list, and/or the ordering of the names? 

So proceedings would be published as `Presenting author, lead devs, other authors, et al for the gammapy team` ?

Author: or authors (PS: a conference contribution can be a poster), lead devs, et al. for the Gammapy collaboration...
A matter of choice;)

Author: I will add examples, good idea.
One should still converge towards an order;)

> followed by 'The Gammapy Collaboration' associated with a link to the Gammapy team webpage.

Reviewer: I agree, gammapy is not really functioning like a collaboration or consortium in the common sense. 
There is a relatively clear gammapy team (regular contributors, maintainers, CC members etc) and less frequent contributors. So in that case I would go for `for the Gammapy team`

If there is a corresponding author, the 'Gammapy Coordination Committee', associated with its usual mailing list, is
used. Concerning the acknowledgement, the Astropy project should always be mentioned.

4. Storage in metadata files
==================================

Depending on the software repository, different metadata files are used in the eco-system of Open Source research
software.

The file ``codemeta.json`` is used by HAL and Software Heritage, and recommended by
`ESCAPE <https://escape2020.pages.in2p3.fr/wp3/eossr/v0.3.2/metadata.html>`_ or
`EOSC <https://www.eosc-pillar.eu/news/illustrating-benefits-eosc-research-communities-germany>`_.

> The file ``citation.cff`` is used by Zenodo and GitHub.

Author: Not necessary!!! Have a look on tributor library. From @vuillaut :
![mermaid-diagram-20220523183724](https://user-images.githubusercontent.com/16781593/171587832-86abb6a0-e64e-41e6-abb3-9e95c4474ded.png)

> The file ``citation.cff`` is used by Zenodo and GitHub.

Reviewer 3: * I'm not an author of tributors
* the schema is simply trying to clarify the current metadata landscape and converters
* `CITATION.cff` may be the reference file, as you can see on the schema it can be the source of conversion to other formats, however, it's a less complete schema than others

5. Possible implementation
==========================

In order to respect these rules, some automation is required to create the list of contributors. One could use
the Python library `Tributors <https://con.github.io/tributors/>`_. This library uses the GitHub API to determine the list of contributors, which is written in the
format associated with codemeta.json, citation.cff, zenodo.json. This kind of library allows then two requirements:
retrieve automatically the list of contributors from GitHub and write the authors list in all the needed formats.

> One should note that the user profile of the GitHub platform does not contain a field associated with the ORCID identifier.

Author: I do not think so. citation.cff will/should be build automatically from mainly the GitHub user profile. However, GitHub does not have the ORCID information. So, one has to ask to people to update the citation.cff file, which is a bit 'dirty'.
This is why I propose to use the ORCID API in the python script that builds the file citation.cff

> One should note that the user profile of the GitHub platform does not contain a field associated with the ORCID identifier.

Reviewer 2: I'm skeptical that this actually works. Names are not unique, can we really associate an ORCID just from the name (and affiliation maybe?)

However, one could also retrieve the ORCID identifier of a contributor with the ORCID API
(e.g. `How-To-1 <https://stackoverflow.com/questions/71052912/extracting-credentials-from-orcid-seach-with-orcid-id-using-python>`_
or `How-To-2 <https://info.orcid.org/documentation/api-tutorials/api-tutorial-searching-the-orcid-registry/#easy-faq-2532>`_).

Even if this library is chosen, a specific Python script located in the main Gammapy repository had to be written such

> that the authors' order follows these rules. In this case, integration within a GitHub action could be possible.

Reviewer 2: Just as a side note, a fist version of such a script is herehttps://github.com/gammapy/gammapy/blob/master/dev/authors.py:

Author: This script is a starting point. But he 'Get list of authors from CITATION.cff', whereas here we wish to create the file author list! So, one should go beyond!

- compliance with the FAIR principles for Research Software (`FAIR4RS <https://www.rd-alliance.org/group/fair-research-software-fair4rs-wg/outcomes/fair-principles-research-software-fair4rs>`_).

This PIG aims to set up the project policy about authorship for our citation scheme that is reminded here.

Given the fact that Gammapy is licensed under a 3-clause BSD style license (see gammapy/LICENSE.rst), Gammapy can be used
and even modified for a science project. For this modified version, the proposed authorship policy of this PIG is not
applicable but the general citation scheme should be applied.

This PIG is structured as follows: a reminder of our general citation scheme that was up-to-now only given orally and on
our web pages; then, the authorship policy is given for each of the products associated with Gammapy, namely the intermediate
releases, the Long Term Support (LTS) releases, the general Gammapy papers, and the contributions to the conference.

2. Citation scheme
==================================

When Gammapy is used for any publication, contribution to conferences or software, authors should cite properly Gammapy.

> It is asked to cite the DOI of the latest Gammapy paper as well as the paper associated to the latest LST release.

Reviewer: LTS

- contributing to our associated projects (gammapy-benchmarks, gammapy-data, gammapy-extra),
- contributing to the maintenance of our repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),
- leading the precise tasks identified by the Lead Developers,
- organizing and leading hands-on sessions or schools,
- managing our social media accounts,
- and obviously coordinating the project by the Lead Developers.

Any contribution without threshold on these topics renders an user as official Gammapy contributor. And the contributions
can be tracked using GitHub. However, the use of personal data is regulated in Europe and one should have the
authorisation of users to be cited with their personal information (typically, their affiliation and their ORCID if any).
For this reason, we propose to use the GitHub `Developer Certificate of Origin <https://github.com/apps/dco>`_ (DCO) on
Pull Request. This lightweight way for contributors will permit to:

- certify that a user wrote or otherwise has the right to submit a code under our open source licence,
- allow our project to use their personal data for the contributions' record (ie for releases).

> The file ``README.md`` will contain a `Contributing` section explaining that the project follows the GitHub DCO, as well

Reviewer: Also in the dev doc?

Author: Yes, as proposed in the PR #4035

who contributed to the current release, as listed in the previous sub-section, who have accepted the GitHub DCO. If the
DCO is not accepted during the Pull Requests, this user will not be in the list of authors per default.

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order:

 ::

  The Gammapy team: aa, bb, cc, ...

As mentioned in the `PIG 23 <https://github.com/gammapy/gammapy/blob/master/docs/development/pigs/pig-023.rst>`_, the
publication of a release is preceded by a code freeze period. With its announcement, a call will be made to contributors
to update their personal information (e.g. their affiliation) under their sole responsibility. This call will also permit
to add any additional author into the list for her/his special contribution. The Project Managers and Lead Developers
will have the duty to examine such request and to update accordingly the author list data . In case of conflict, the
Gammapy Coordination Committee will be the final decision maker.

Long Term Support releases
--------------------------

> The list of authors is composed of the union of all the contributors of the releases realised since the last LST

Reviewer: LTS

release. As an LTS will be associated with a paper, the members of the Coordination Committee will appear in this
peer-review product as a long-term support of the Gammapy project (see next section).

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order.

Like for the common releases, the period of code freeze is used to make a call to update the personal information of
contributors, as well as to submit a request of co-authorship for special contributions. The same rules and methods
as for the standard releases are applied here.

> For the first LST release, the `V1.0`, all contributors from the beginning of the project will be co-authors per default.

Reviewer: LTS

contributions. The order of the authors is 'The Gammapy team', followed by the list of contributors in alphabetical order.

General Gammapy publications
----------------------------

This product aims to describe the project and/or the library as a whole and will be more of the case associated with the
publication of an LTS release. As it targets a wide community, the following scheme is used.

1. List of authors
    a/ By default, 'The Gammapy team' is mentioned first, then the Lead Developers, then the past Lead Developers since
    the last general Gammapy publication, then the list of contributors of each release since the last Gammapy publication.

    b/ If the editorial rules of the targeted journal permit it, the scheme used by the Astropy project
    (e.g. `Astropy V2.0 <https://arxiv.org/pdf/1801.02634.pdf>`_) should be used in priority:

    - 'The Gammapy team', and the list of primary paper contributors in order of contribution agreed per consensus with the development team, as '(Primary Paper Contributors)',
    - the members of the Coordination Committee, as '(Gammapy Coordination Committee)',

>     - the list of contributors of the associated LST ordered by alphabetical order, as '(Gammapy Contributors)'

Reviewer: LTS

will have the duty to examine such request and to update accordingly the author list data . In case of conflict, the
Gammapy Coordination Committee will be the final decision maker.

Long Term Support releases
--------------------------

The list of authors is composed of the union of all the contributors of the releases realised since the last LTS
release. As an LTS will be associated with a paper, the members of the Coordination Committee will appear in this
peer-review product as a long-term support of the Gammapy project (see next section).

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order.

Like for the common releases, the period of code freeze is used to make a call to update the personal information of
contributors, as well as to submit a request of co-authorship for special contributions. The same rules and methods
as for the standard releases are applied here.

For the first LTS release, the `V1.0`, all contributors from the beginning of the project will be co-authors per default.
As the DCO has not be applied for the past contributions, the best will be done to contact all contributors in order to

> exercise their right to withdraw (OptOut scheme), to update their personal information, and to add authors with special

Reviewer 2: ```suggestion
exercise their right to withdraw ("opt out" scheme), to update their personal information, and to add authors with special
```

the Gammapy project itself. It does not concern a technical or scientific work that uses Gammapy as an open library.
In this case, the citation scheme of Gammapy should be used by these authors.

As the length of the author list is generally a constrain, the author list is reduced to the short list of contributors
for the conference, followed by 'The Gammapy team' associated with a link to the Gammapy team webpage:

 ::

  oo, ff, tt, for `the Gammapy team <https://gammapy.org/team.html>`_

If there is a corresponding author, the 'Gammapy Coordination Committee' associated with its usual mailing list, is
used. Concerning the acknowledgement, the Astropy project should always be mentioned, and if possible our mandatory
external libraries.

4. Metadata files
==================================

Depending on the software repository, different metadata files are used in the eco-system of Open Source research
software.

The file ``codemeta.json`` is used by HAL and Software Heritage, and recommended by
`ESCAPE <https://escape2020.pages.in2p3.fr/wp3/eossr/v0.3.2/metadata.html>`_ or
`EOSC <https://www.eosc-pillar.eu/news/illustrating-benefits-eosc-research-communities-germany>`_.

> The file ``CITATION.cff`` is used by Zenodo and GitHub. Its format should follow the rules set up by the

Reviewer 2: I think there should be as statement, that the `CITATION.cff` will be used as the main file to be maintained, while the `codemeta.json` will be automatically created. The `CITATION.cff` is the preferred maintenance file, because it is human readable.

Author: done

string, one could add a past affiliation as precised below, but with a risk that Zenodo or tools using ``CITATION.cff``
to make the ``codemeta.json`` does not understand it.

 ::

    authors:
      - family-names: XXX
        given-names: "YYYY Z."
        affiliation: "Lab AA, XX; Past: Lab BB, YY"
        email: yyyy.xxx@gammapy.org
        orcid: "https://orcid.org/0000-0000-0000-0000"

For any release, both metadata files should be carefully updated and systematically a review should be done by the dev
team.

5. Possible implementation
==========================

In order to respect these rules, some automation is required to create the list of contributors. One could use
the Python library `Tributors <https://con.github.io/tributors/>`_. This library uses the GitHub API to determine the
list of contributors, which is written in the format associated with codemeta.json, citation.cff, zenodo.json. This kind
of library allows then two requirements: retrieve automatically the list of contributors from GitHub and write the
authors list in all the needed formats.

One should note that the user profile of the GitHub platform does not contain a field associated with the ORCID

> identifier. However, one could also retrieve the ORCID identifier of a contributor with the ORCID API

Reviewer 2: I think we will just ask for there ORCID from contributors as we do for the affiliation. The ORCID typically does not change and it will be stored and maintained in the `CITATION.cff`...

Author: There is a small subtlety: the citation file contains the list of contributors since the latest feature release, not from all the beginning of the project. So the ORCID id can not be stored into this file for all users.

Reviewer 2: Thanks for clarifying, indeed this I have overseen! This raises the question for me whether we maybe introduce an independent `AUTHORS.yaml` for maintenance which contains the author entries independent of the current release. This could be used for LTS releases as well. What do you think?

Author: This could technically help and help the maintenance. To avoid the confusion with the CITATION.cff for external person, one could call it `LTS_AUTHORS.yaml`. 
I am pro this concept... and if you are with the name, I go for it both in this PR and in the PR that I prepare about the script `dev/authors.py`

Reviewer 2: Sounds good to me. I think having an author list "archive" with information independent of any version will help maintenance, because we can create the "running" author list in `CITATION.cff` from it.

Reviewer 4: I find it unusual to drop people from the citation. In my opinion, the CITATION.cff should include all people who have contributed to Gammapy by the definitions above, not just the ones contributing between the last and the current release.

In the extreme example, a minor version release with a single bugfix would have a single author. That's clearly not wanted, is it?

That also makes it possible to use this file as the one source of truth about the contributors where they can provide there own data themselves via pull requests, which solves the consent problem.

For past contributors, we could write them to give there consent explicitly in the first update to the citation.cff we do before the next release.

Author: They should be a mis-understanding as nobody is dropped, unless under her/his request.
Otherwise, I add a sentence to be clear on the bug release.

Reviewer 2: I think whether dropping people from author lists is fine or not depends on the time scale. E.g. it is very common for large collaborations that people are removed from the author list once they have left the collaboration since a while. To my knowledge the policy proposed in the PIG is:
```
The list of authors is composed of the union of all the contributors of the releases realised since the last LTS release.
```
This means the `CITATION.cff` is a "running" author list which will consist of all people that contributed since the last LTS release. This  defines a timeline of updating the author list of ~2 years, which I think is fair. The case that you just mentioned with a "single author" cannot happen. On the other hand "old contributors" that have left the project before v1.0 will not be part of the v2.0 author list.

The "all time" contributor list will then be an independently manually maintained `ALL_AUTHORS_EVER.yaml` file. From which the `CITATION.cff` (which is e.g. used for Zenodo) is created automatically. I hope this makes sense.

Reviewer 2: Addendum: v1.0 will include all contributors since the beginning of the project.

Reviewer 2: Ok, I know realize that indeed bug fix releases need to be specified more precisely. Taking the policy very strictly would indeed mean that `v1.0.1` would only include the people that have contributed to the bug-fix release, which does not seem fair. I guess the better default for bug fix release would be the authors from the release the bug-fix relates too plus additional new authors that e.g. have contributed the bug fix.

Author: Indeed, I have just changed it after one comment of Max in this way ;)

- compliance with the FAIR principles for Research Software (`FAIR4RS <https://www.rd-alliance.org/group/fair-research-software-fair4rs-wg/outcomes/fair-principles-research-software-fair4rs>`_).

This PIG aims to set up the project policy about authorship for our citation scheme that is reminded here.

Given the fact that Gammapy is licensed under a 3-clause BSD style license (see gammapy/LICENSE.rst), Gammapy can be used
and even modified for a science project. For this modified version, the proposed authorship policy of this PIG is not
applicable but the general citation scheme should be applied.

This PIG is structured as follows: a reminder of our general citation scheme that was up-to-now only given orally and on
our web pages; then, the authorship policy is given for each of the products associated with Gammapy, namely the intermediate
releases, the Long Term Support (LTS) releases, the general Gammapy papers, and the contributions to the conference.

2. Citation scheme
==================================

When Gammapy is used for any publication, contribution to conferences or software, authors should cite properly Gammapy.

> It is asked to cite the DOI of the latest Gammapy version as well as the paper associated to the latest LTS release.

Author: Thanks Max, a typo. 2 citations should be needed: the DOI of the used version of Gammapy and the latest Gammapy paper.

Definition of a `Contributor`
-----------------------------

Contributing to Gammapy can be made in different ways:

- contributing to gammapy source code,
- contributing to our documentation (rst pages, docstring, gammapy-webpage),
- contributing to our DevOps environment (e.g. CI, docs and package deployment, GitHub configuration),
- contributing to our associated projects (gammapy-benchmarks, gammapy-data, gammapy-extra),
- contributing to the maintenance of our repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),
- leading the precise tasks identified by the Lead Developers,
- organizing and leading hands-on sessions or schools,
- managing our social media accounts,
- and obviously coordinating the project by the Lead Developers.

Any contribution without threshold on these topics renders an user as official Gammapy contributor. And the contributions
are tracked using the GitHub contributions. However, the use of personal data is regulated in Europe and one should have the
authorisation of users to be cited with their personal information (typically, their affiliation and their ORCID if any).

> For this reason, we propose to use the GitHub `Developer Certificate of Origin <https://github.com/apps/dco>`_ (DCO) on

Reviewer 2: I agree a "one time sign off" solution would be much preferable...

> For this reason, we propose to use the GitHub `Developer Certificate of Origin <https://github.com/apps/dco>`_ (DCO) on

Author: This PR settles the DCO signature... The implementation is made in the DCO PR

releases, the Long Term Support (LTS) releases, the general Gammapy papers, and the contributions to the conference.

2. Citation scheme
==================================

When Gammapy is used for any publication, contribution to conferences or software, authors should properly cite Gammapy.
It is asked to cite the DOI of the used Gammapy version as well as the associated paper (e.g. the latest LTS
release).

.. note::
    As the Gammapy community fully supports Open Science, we strongly encourage authors using Gammapy to follow
    the FAIR4RS principles and to allow the reproducibility of their results. As consequence, we suggest always mentioning
    the `Zenodo <https://zenodo.org/>`_ DOI or the `HAL <https://hal.archives-ouvertes.fr/>`_ SWHID identifier (associated with
    the `Software Heritage <https://www.softwareheritage.org/>`_ archive) of the used release of Gammapy.

This citation scheme with these **two references** will be given on our web pages.

3. Authorship policy
====================

The Gammapy references contain a list of authors that requires to be updated with time according to a

> general policy. This section defines what is a `contributor` to Gammapy and the policy for each type of product.

Reviewer 4: Do we anticipate non-human contributors? The list below doesn't look like it. If contributors can only be human, I'd write

> defines *who* is a contributor

Definition of a `Contributor`
-----------------------------

> Contributing to Gammapy can be made in different ways:

Reviewer 4: Contributions to Gammapy can be made ...

or

Contributing to Gammapy is possible ...

- contributing to gammapy source code,
- contributing to our documentation (rst pages, docstring, gammapy-webpage),
- contributing to our DevOps environment (e.g. CI, docs and package deployment, GitHub configuration),
- contributing to our associated projects (gammapy-benchmarks, gammapy-data, gammapy-extra),
- contributing to the maintenance of Gammapy products on open-source repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),
- leading the precise tasks identified by the Lead Developers,
- organizing and leading hands-on sessions or schools,
- managing our social media accounts,

> - and coordinating the project by the Lead Developers.

Reviewer 4: This sentence does not really fit the "what makes a person a contributor" I think... Lead Developers will by definition always be contributors, because it's expected of lead developers that they do many of the things listed above.

Maybe we can summarize the different a bit vague points like the one about the "precise tasks", the "lead developers" and thinks like website and social media to a single point

> taking an active role in coordinating or promoting the project

?

Any contribution without threshold on these topics renders a user as official Gammapy contributor. And the contributions

> are tracked using the GitHub contributions. However, the use of personal data is regulated in Europe and one should have the

Reviewer 4: Concerning 

> And the contributions are tracked using the GitHub contributions

We list many things above that are not directly track-able using GitHub (e.g. zenodo, coordinating, website, etc...) or harder than just looking at contributions here (conda-forge, ...),  so I'd leave this out.

Author: Indeed, the sentence is too strict. The -> Many

authorisation of users to be cited with their personal information (typically, their affiliation and their ORCID if any).
For this reason, we propose to use the GitHub `Developer Certificate of Origin <https://github.com/apps/dco>`_ (DCO) on
Pull Request. Its acceptation by contributors will permit to:

- certify that a user wrote or otherwise has the right to submit a code under our open source licence,
- allow our project to use their personal data for the contributions' record (ie for releases).

The file ``README.md`` will contain a `Contributing` section explaining that the project follows the GitHub DCO, as well
as a link to this PIG.

.. important::
    We strongly recommend that the contributors use always the same GitHub account with a valid full name, to

>     synchronise their git email address with their GitHub address, to have a unique `ORCID <https://orcid.org/>`_

Reviewer 4: plural, email addresses. Both github and git itself allow usage of multiple email addresses

>     identifier and store it into their GitHub user profile for commodity, and inform the developers of any change of

Reviewer 4: As discussed in Slack and more details below, we should not rely on people's GitHub user profiles.

Rather we should ask people to contribute the information they want to have listed for their Gammapy contributions here, which also would directly provide the explicit consent necessary for the usage of the data

Author: We are not relying on them, but we will use them. A call is made to authors, as described in the sections of releases, papers and conferences

Reviewer 2: I think there is a misunderstanding. We will not use GitHub profile data in an automatized way, but it can be a way to reach out to people to collect the data for the first `ALL_AUTHORS_EVER.yaml` file. So we will send around a request, containing a link to a Google form or request to open a PR to everyone who is in the git history of Gamapy up to now. In individual cases we might look at their GitHub profiles to get e.g. an update to date mail address to send the request to. 
Once this first `ALL_AUTHORS_EVER.yaml` is theere (hopefully complete based on voluntary filling of the Google form...) we will just maintain it by hand. New contributors will add their information or once an affiliation changes people need to submit a PR to update it.

The only controversial thing to me is, how to get this first `ALL_AUTHORS_EVER.yaml` as complete as possible. In this case we might announce to use the name of the people (no affiliation, no ORCID, no mail-address) if they do not object. In future this will happen via the DCO.

    affiliation.

Definition of the `Maintainer`
------------------------------

In the Open Source community, a maintainer is generally mentioned in the metadata associated with a release and has to be mentioned.

> Usually, a software maintainer or package maintainer is one or more people who build source code into a binary package

Reviewer 4: Language: *a* maintainer cannot be multiple people. Just define the term and mention that there could be multiple maintainers.

Reviewer 4: This was not changed as far as I can see.

for distribution, commit patches, or organize code in a source repository.

For Gammapy, by default, the maintainers are the Lead Developers. If in the future a task is dedicated to the creation
of a release, the maintainer will be the person in charge of this task.

Releases
--------

Each Gammapy release should be associated with an updated list of authors that will be public on repository (Zenodo and

> HAL/SW). This section is about feature or bug releases, not the LTS release. The list of authors is composed by people

Reviewer 4: > This section is about feature or bug releases, not the LTS release.

What is  different for LTS releases and why? 

> The list of authors is composed by

composed of

> who contributed to the current release, as listed in the previous sub-section, who have accepted the GitHub DCO. If the

Reviewer 4: The previous subsection deals with "Maintainers" not "Contributors"

DCO is not accepted by a contributor, her/his contribution might not be usable and thus this user will not be in the
list of authors per default.

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order:

 ::

  The Gammapy team: aa, bb, cc, ...

As mentioned in the `PIG 23 <https://github.com/gammapy/gammapy/blob/master/docs/development/pigs/pig-023.rst>`_, the
publication of a release is preceded by a code freeze period. With its announcement, a call will be made to contributors
to update their personal information (e.g. their affiliation) under their sole responsibility. This call will also permit
to add any additional author into the list for her/his special contribution. The Project Managers and Lead Developers
will have the duty to examine such request and to update accordingly the author list data . In case of conflict, the
Gammapy Coordination Committee will be the final decision maker.

Long Term Support releases
--------------------------

The list of authors is composed of the union of all the contributors of the releases realised since the last LTS

> release. As an LTS will be associated with a paper, the members of the Coordination Committee will appear in this

Reviewer 4: As an LTS *release*

peer-review product as a long-term support of the Gammapy project (see next section).

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order.

> Like for the common releases, the period of code freeze is used to make a call to update the personal information of

Reviewer 4: a period

`EOSC <https://www.eosc-pillar.eu/news/illustrating-benefits-eosc-research-communities-germany>`_.

The file ``CITATION.cff`` is used by Zenodo and GitHub. Its format should follow the rules set up by the
`Citation File Format project <https://pythonlang.dev/repo/citation-file-format-citation-file-format/>`_. A GitHub
Action can be used to automatically check the compliance with the latest format. At the date of this PIG, there is no
official scheme to handle current and past affiliations, that might be needed for LTS releases for example (see the
CFF project `Issue #268 <https://github.com/citation-file-format/citation-file-format/issues/268>`_). In the affiliation
string, one could add a past affiliation as precised below, but with a risk that Zenodo or tools using ``CITATION.cff``
to make the ``codemeta.json`` does not understand it.

 ::

    authors:
      - family-names: XXX
        given-names: "YYYY Z."
        affiliation: "Lab AA, XX; Past: Lab BB, YY"
        email: yyyy.xxx@gammapy.org
        orcid: "https://orcid.org/0000-0000-0000-0000"

The ``CITATION.cff`` file will be used as the main file to be maintained, while the ``codemeta.json`` will be

> automatically created. For any release, the former should be carefully updated and systematically a review should be

Reviewer 4: Strike systematically. I'd also switch the order. One first reviews and then updates if necessary

of library allows then two requirements: retrieve automatically the list of contributors from GitHub and write the
authors list in all the needed formats.

One should note that the user profile of the GitHub platform does not contain a field associated with the ORCID
identifier. However, one could also retrieve the ORCID identifier of a contributor with the ORCID API
(e.g. `How-To-1 <https://stackoverflow.com/questions/71052912/extracting-credentials-from-orcid-seach-with-orcid-id-using-python>`_
or `How-To-2 <https://info.orcid.org/documentation/api-tutorials/api-tutorial-searching-the-orcid-registry/#easy-faq-2532>`_).

Even if this library is chosen, a specific Python script located in the main Gammapy repository had to be written such
that the authors' order follows these rules. In this case, integration within a GitHub action could be possible.

Finally, a second script should be settled in order to create the list of contributors in HTML format that should be
inserted in the `team` section for our general web page. This script would read one of the metadata files (e.g.
``citation.cff``) to create such list.

> Concerning the contributions to conferences, their handling should be realised properly by creating a dedicated

Reviewer 4: This is vague. What does "properly realizing the handling" mean? What are we actually expecting here?

E.g. would:

> Conference contributions (Proceedings, Posters, ...) should be developed in dedicated repositories in the Gammapy Github organization?

be more clear of the intent?

repository under the Gammapy organisation of GitHub.

6. Suggestions
==============

One could ask CTAO software coordinators (ie ACADA, DPPS, SUSS) if such rules can be used by them also, even if the
The Gammapy project is independent. In the same spirit, advice could be asked to people of the ESCAPE project via our

> corresponding person. And finally, one could mention the existence of these rules and ask for advice to the Astropy

Reviewer 4: Shouldn't this, especially the "Asking the astropy project for advice" be done first? They have  a much longer tradition as an independent (aka no single entity behind it) open source project.

Reviewer 2: I actually asked on the Astropy Slack without getting much response. I haven't even received a clear answer on how their Zenodo author list is created. So it seems there is no policy in place form their side. There is no APE about this either.

Author: For CTAO, I have asked... no clear answer from the management. They will see this later

Reviewer 2: @bkhelifi Let's write this in the PIG e.g. like "Requests to the Astropy community and CTAO for recommendations and preferences remained un-responded at the current time." or similar...

    identifier and store it into their GitHub user profile for commodity, and inform the developers of any change of
    affiliation.

Definition of the `Maintainer`
------------------------------

Usually, a software maintainer or package maintainer is one or more people who build source code into a binary package
for distribution, commit patches, or organize code in a source repository. In the Open Source community, they are
generally mentioned in the metadata associated with a release.

For Gammapy, by default, the maintainers are the Lead Developers. If in the future a task is dedicated to the creation
of a release, the maintainer will be the person in charge of this task.

Releases
--------

Each Gammapy release should be associated with an updated list of authors that will be public on repository (Zenodo and
HAL/SW). This section is about feature releases. Bug releases will have as list of authors the one of its associated
feature release whose rules are described here, plus the potentiel additional contributors.

The list of authors is composed of people who contributed to the current release (see :ref:`_contributor`) and who have

> accepted the GitHub DCO. If the DCO is not accepted by a contributor, her/his contribution might not be usable and thus

Reviewer 4: The "Gammapy Contribution Terms" or similar, the actual content should be defined in this PIG.

The DCO comes from the Linux and Git projects, not Github. The app you link is a thirdparty app, also not from Github.

Using signed-off-by does not relate directly to the specific DCO of Linux and Git, it is defined for each project individually, see the git help (`man git commit`):

> Add a Signed-off-by trailer by the committer at the end of the commit log message. The meaning of a signoff depends on the
> project to which you’re committing. For example, it may certify that the committer has the rights to submit the work under the
> project’s license or agrees to some contributor representation, such as a Developer Certificate of Origin. (See
> http://developercertificate.org for the one used by the Linux kernel and Git projects.) Consult the documentation or leadership
> of the project to which you’re contributing to understand how the signoffs are used in that project.

It mentions the git and linux DCOs as one possibibility, but projects have to define their own policy. We could define that we use the same or make our own with additional or fewer terms, but we need to define it (here and also I think in the Pull Request template and `CONTRIBUTING.md` and the developer docs).

I think the Linux and Git DCO is a good starting point, but we could add things specific to gammapy, e.g. mention the usage of the PII for publication author lists.

Author: Corrections and addition made

this user will not be in the list of authors per default.

The order of the authors is first 'The Gammapy team', followed by the list of contributors in alphabetical order:

 ::

  The Gammapy team: aa, bb, cc, ...

As mentioned in the `PIG 23 <https://github.com/gammapy/gammapy/blob/master/docs/development/pigs/pig-023.rst>`_, the
publication of a release is preceded by a code freeze period. With its announcement, a call will be made to contributors
to update their personal information (e.g. their affiliation) under their sole responsibility. This call will also permit

> to add any additional author into the list for her/his special contribution. The Project Managers and Lead Developers

Reviewer 4: The his/her is in most modern English text now replaced by just using `they` or `them`, which eases the reading I think and proponents argue does not exclude non-binary people. 

https://en.wikipedia.org/wiki/Singular_they

Reviewer 4: Actually, the sentence above already makes use of it (e.g. their sole responsibility), so let's just be consistent.

    In this case, a comment on the author list composition should be added. Extracted from the Astropy project,
    one can precise:

    "The author list has three parts: the authors that made significant contributions to the writing of the paper,
    the members of the Gammapy Project Coordination Committee, and contributors to the Gammapy Project in alphabetical
    order. The position in the author list does not correspond to contributions to the Gammapy Project as a whole. A more
    complete list of contributors to the core package can be found in the package repository and at the Gammapy team
    webpage."

2. Corresponding author
    It is the 'Gammapy Coordination Committee', associated with its usual mailing list (GAMMAPY-COORDINATION-L@IN2P3.FR).

3. Acknowledgement
    One has the freedom to precisely mention acknowledgements associated with the publication. In practice, it is
    recommended to precise the grants or fellowships given to some authors. One should in any case always acknowledge
    the Astropy project, to which we are affiliated, and our mandatory external libraries (e.g. numpy, scipy, matplotlib,

>     iminuit).

Reviewer 4: This is probably a separate issue, but should there be a general Gammapy Acknowledgement? E.g. on the home page?

E.g. my funding agency does in principle require this, and I guess this holds true for many other people

Reviewer 2: I agree we could mention here the possibility of maintaining an independent list of acknowledgments e.g. on the webpage.

 ::

  oo, ff, tt, for `the Gammapy team <https://gammapy.org/team.html>`_

If there is a corresponding author, the 'Gammapy Coordination Committee' associated with its usual mailing list, is
used. Concerning the acknowledgement, the Astropy project should always be mentioned, and if possible our mandatory
external libraries.

4. Metadata files
==================================

Depending on the software repository, different metadata files are used in the eco-system of Open Source research
software and are mandatory.

The file ``codemeta.json`` is used by HAL and Software Heritage, and recommended by
`ESCAPE <https://escape2020.pages.in2p3.fr/wp3/eossr/v0.3.2/metadata.html>`_ or
`EOSC <https://www.eosc-pillar.eu/news/illustrating-benefits-eosc-research-communities-germany>`_.

The file ``CITATION.cff`` is used by Zenodo and GitHub. Its format should follow the rules set up by the
`Citation File Format project <https://pythonlang.dev/repo/citation-file-format-citation-file-format/>`_. A GitHub
Action can be used to automatically check the compliance with the latest format. At the date of this PIG, there is no

> official scheme to handle current and past affiliations, that might be needed for LTS releases for example (see the

Reviewer 4: I don't understand why an LTS release needs current or past affiliations. LTS releases are made from specific branches in the development scheme. Each of this branches needs to have the affiliation for that LTS release in the version of the `CITATION.cff` in that branch, then the tag (release) is made.

E.g. the workflow for 1.0 would be:

* Announce code-freeze
* Create the v1.0.x branch
* Ask people to update the CITATION.cff on the v1.0.x branch
* Make the v1.0.0 tag

Correct?

Reviewer 2: I think the issue here is that there is no support for multiple affiliations in `CITATION.cff` (see https://github.com/citation-file-format/citation-file-format/issues/268), however for some cases it might be required, because contributors can be affiliated to multiple institutions at the same time. Or their Gammapy contributions were made while being affiliated to multiple institutes (which might be very common for post-docs, changing affiliations after 2 years).

I think the only simple solution here is that week ask people to put their "preferred" affiliation and make a choice. So if a person contributed over the time span of 2 years, which ends with an LTS release and changed affiliation in between then they have to specify which one they prefer to be listed e.g. on Zenodo. For associated papers their might be the possibility to extract the information (change of affiliation) from the git history and list multiple if needed.

Reviewer 4: Citations in general don't really have this feature (bibtex doesn't even have a field for authors affiliations). So the only solution to this would be to ask people to contribute this information to the actual paper author list in the corresponding repo or collect this information via some survey for any specific paper.

This is what astropy did: call for paper authors to provide their information via a google form

releases, the Long Term Support (LTS) releases, the general Gammapy papers, and the contributions to the conference.

2. Citation scheme
==================================

When Gammapy is used for any publication, contribution to conferences or software, authors should properly cite Gammapy.
It is asked to cite the DOI of the used Gammapy version as well as the associated paper (e.g. the latest LTS
release).

.. note::
    As the Gammapy community fully supports Open Science, we strongly encourage authors using Gammapy to follow
    the FAIR4RS principles and to allow the reproducibility of their results. As consequence, we suggest always mentioning
    the `Zenodo <https://zenodo.org/>`_ DOI or the `HAL <https://hal.archives-ouvertes.fr/>`_ SWHID identifier (associated with
    the `Software Heritage <https://www.softwareheritage.org/>`_ archive) of the used release of Gammapy.

This citation scheme with these **two references** will be given on our web pages.

3. Authorship policy
====================

The Gammapy references contain a list of authors that requires to be updated with time according to a

> general policy. This section defines who is a `contributor` to Gammapy and the policy for each type of product.

Reviewer 4: Shouldn't it be just normal quotes? Syntax Highlighting doesn't make sense here

> general policy. This section defines who is a `contributor` to Gammapy and the policy for each type of product.

Reviewer 2: Yes, sphinx will try to render \`contributor\` as a code / API link. It will show up as warning in the docs build. So just use normal quotes...

5. Possible implementation
==========================

5.1 DCO implementation
----------------------

Today, the tool offered by GitHub to sign the DCO is to add the extra parameter ``-s`` to each git commit. Users have to
be aware of that and get used of it. For the reviewers, a github CI can be used to check quickly if a contributor has
signed the DCO. This method will be used for the moment. If there is any other method that avoids this extra parameter,
one will naturally consider it.

In order to respect these rules, some automation is required to create the list of contributors. One could use
the Python library `Tributors <https://con.github.io/tributors/>`_. This library uses the GitHub API to determine the
list of contributors, which is written in the format associated with codemeta.json, citation.cff, zenodo.json. This kind
of library allows then two requirements: retrieve automatically the list of contributors from GitHub and write the
authors list in all the needed formats.

5.2 Collection of the personal information of authors
-----------------------------------------------------

> The data collected from the git logs are not sufficient for any publication. Most of the contributors must give their

Reviewer: not only this but also contributions might not belong to the git history.

affiliation as stipulated by their working contract. Also, the ORCID number is a relatively new type of information,
recommended to be used in the context of the Open Science movement, but it is not mandatory. However, one needs to
collect this information to build the authors lists, in a safe and fair way.

In this purpose, one could use the public user profile of the GitHub platform. They are public data, but one should

> aware that they are not complete and/or up-to-date. As mention earlier, it is recommended to fill correctly this GitHub

Reviewer: should be aware

Reviewer: mentioned

> profile to help the dev team to pre-filled the metadata files.

Reviewer: pre-fill

One should note that the GitHub profile does not contain a field associated with the ORCID
identifier. However, one could also retrieve the ORCID identifier of a contributor with the ORCID API
(e.g. `How-To-1 <https://stackoverflow.com/questions/71052912/extracting-credentials-from-orcid-seach-with-orcid-id-using-python>`_
or `How-To-2 <https://info.orcid.org/documentation/api-tutorials/api-tutorial-searching-the-orcid-registry/#easy-faq-2532>`_).

In any case, a specific Python script located in the main Gammapy repository had to be written such
that the authors' order follows these rules. In this case, integration within a GitHub action could be possible.

In this context, one could technically use an other cff file as maintenance file to store the personal data of users
that have signed the DCO.

> A second script should be settled in order to create the list of contributors in HTML format that should be

Reviewer: How frequently is the team description planned to be updated? What is on the web page might be different from the releases/papers?

Reviewer: Note that this list should not contain any personal data other than the name I suppose

inserted in the `team` section for our general web page. This script would read one of the metadata files (e.g.
``citation.cff``) to create such list.

It is important to note that the list of personal information might change from one editor to an other. For the code
deposits in Zenodo, SWH, etc, the data follow standards of Open Science and scripts can be settled. For publications,
the editors have frequently their own scheme and adaptations will be made on the case by case.

As mentioned here, the authors list will be reviewed for any publication, for safety and to allow OptIn/OptOut requests.

5.3 Handling of conference material
-----------------------------------

Conference contributions (Proceedings, Posters, ...) could be developed in dedicated repositories in the Gammapy Github
organization, as well as the author list.

> 6. Suggestions

Reviewer: numbering

==============

One could ask CTAO software coordinators (ie ACADA, DPPS, SUSS) if such rules can be used by them also, even if the
The Gammapy project is independent. In the same spirit, advice could be asked to people of the ESCAPE project via our
corresponding person. And finally, one could mention the existence of these rules and ask for advice to the Astropy
project.

These requests to the Astropy community and CTAO for recommendations and preferences either remained
un-responded or did not lead to objections at the current time.

7. Decision
===========

> After the addition of comments and proposals, the PIG is accepted by the dev team and the CC.

Reviewer: Maybe add a sentence about the choice of practical implementation to be decided later?

.. include:: ../../references.txt

.. _pig-024:

**************************
PIG 24 - Authorship policy
**************************

* Authors: Bruno Khélifi, Thomas Vuillaume
* Created: May 25th, 2022

> * Accepted: Sept. ?th, 2022

Reviewer: ```suggestion
* Accepted: Oct. 20th, 2022
```

> * Status:

Reviewer: ```suggestion
* Status: accepted
```

> 

Reviewer: ```suggestion
* Discussion: `GH 3970`_

```

information for publications (their full name is mandatory, and, if any, their affiliation and
their ORCID number). For this reason, we propose to use the *Developer Certificate of Origin*
(DCO, see the text in `here <https://developercertificate.org/>`_) for each commit. Its
acceptation by contributors will permit to:

* certify that a user wrote or otherwise has the right to submit a code under our open source licence,
* allow our project to use their personal data for the contributions' record (ie for releases).

The file ``README.md`` will contain a *Contributing* section explaining that the project follows this DCO that will be
given in a new ``CONTRIBUTING.md`` file, as well as a link to this PIG.

.. important::
    For practical reasons, it is strongly advised that the contributors use always the same GitHub account
    with a valid full name, to synchronise their git email addresses with their GitHub addresses, to have a unique
    `ORCID <https://orcid.org/>`_ identifier and store it into their GitHub user profile for commodity, and inform the
    developers of any change of affiliation or email address.

> Definition of the *Maintainer*

Reviewer 4: Why *the* ? We have multiple maintainers, right?

I'd just leave out the article (for both Contributor and Maintainer):

> Definition of Maintainer

Reviewer 2: I also support consistency here, I think either `Definition of Maintainer` or `Definition of a Maintainer` (consistent with the contributor section right now) would be fine by me.

- contributing to gammapy source code,
- contributing to our documentation (rst pages, docstring, gammapy-webpage),
- contributing to our DevOps environment (e.g. CI, docs and package deployment, GitHub configuration),
- contributing to our associated projects (gammapy-benchmarks, gammapy-data, gammapy-extra, etc),
- contributing to the maintenance of Gammapy products on open-source repositories (conda-forge, PyPI, Zenodo, HAL, docker-hub, etc) or open hub (openhub, etc),
- leading the precise tasks identified by the Lead Developers (e.g module maintainer),
- organizing and leading hands-on sessions or schools,
- managing our social media accounts,
- and coordinating the project by the Lead Developers.

Any contribution without threshold on these topics renders a user as official Gammapy contributor. Many of these
contributions are tracked using the git history. However, the use of personal data is regulated in Europe and one

> should have the authorisation of users to be cited with their personal information for publications (their full name

Reviewer 4: Their full name *is* mandatory

> are mandatory, and if any their affiliation and their ORCID number). For this reason, we propose to use the

Reviewer 4: optionally, we collect affiliations and ORCID numbers

Author: The use of 'if any' aims to promote this ORCID id. We aim to push for that as much as possible. In the case of the existence of the ORCID for a user, we gonna to use it.

Reviewer 2: I also agree we should encourage the people to provide the ORCID if present. It is in their own interest and meta data will be more complete...

Reviewer 4: This was mainly a language comment, it read a bit awkward. 

Maybe:

> (the full name is mandatory and we highly encourage also providing affiliation and ORCID)

?

Author: awkward? In which sense? how the wording 'if any' can be awkward?

Reviewer 2: @bkhelifi I guess surrounding commas would make a difference as well. Like
```
are mandatory, and, if any, their affiliation and their ORCID number
```

Reviewer 5: full name *is* mandatory
